packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignInMethodEditBox/AddSignInMethodButton.tsx

```diff
@@ -0,0 +1,64 @@
+import { SignInIdentifier } from '@logto/schemas';
+import { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { snakeCase } from 'snake-case';
+
+import Button from '@/components/Button';
+import Dropdown, { DropdownItem } from '@/components/Dropdown';
+
+type Props = {
+  options: SignInIdentifier[];
+  onSelected: (signInIdentifier: SignInIdentifier) => void;
+};
+
+// TODO: @yijun extract this component to share with the future add social button
+const AddSignInMethodButton = ({ options, onSelected }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const anchorRef = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const candidates = options.map((identifier) => ({
+    value: identifier,
+    title: t('sign_in_exp.sign_up_and_sign_in.identifiers', { context: snakeCase(identifier) }),
+  }));
+
+  if (candidates.length === 0) {
+    return null;
+  }
+
+  return (
+    <>
```

Comment:
nit: div

Reply:
Maybe not?
The element after the `Button` is a `<Dropdown />`, which is a modal, and will not come after the button element in the document flow.

Reply:
We have a related issue: https://linear.app/silverhand/issue/LOG-4282/forward-reference-for-the-button-component-and-refactor-related-code

Will refactor it all in the code base.